internal/querynode/distribution.go

```diff
@@ -208,10 +207,11 @@ func (d *distribution) RemoveDistributions(releaseFn func(), sealedSegments ...S
 			delete(d.sealedSegments, sealed.SegmentID)
 		}
 	}
-
 	d.offlines.Add(delta)
-
 	<-d.genSnapshot()
+
+	d.mut.Unlock()
+
 	releaseFn()
```

Comment:
`releaseFn` shall be executed before lock released

Reply:
Why? I didn't see any distribution related code in release fn

Reply:
to prevent incoming search reading any possible growing segments need to be released